pkg/controller/servicebindingrequest/binder.go

```diff
@@ -501,32 +513,39 @@ func (b *binder) update(objs *unstructured.UnstructuredList) ([]*unstructured.Un
 
 		sbrNamespacedName := types.NamespacedName{Namespace: b.sbr.GetNamespace(), Name: b.sbr.GetName()}
 		updatedObj = setSBRAnnotations(sbrNamespacedName, updatedObj)
+		var err error
+		if b.sbr.Spec.ApplicationSelector.BindingPath.SecretPath != "" {
+			err = b.updateSecretField(updatedObj)
+			if err != nil {
+				return nil, err
+			}
+		}
 
-		updatedObj, err := b.updateSpecContainers(updatedObj)
-		if err != nil {
-			return nil, err
+		if b.sbr.Spec.ApplicationSelector.BindingPath.ContainersPath != "" {
+			err = b.updateSpecContainers(updatedObj)
+			if err != nil {
+				return nil, err
+			}
 		}
 
 		if len(b.volumeKeys) > 0 {
-			if updatedObj, err = b.updateSpecVolumes(updatedObj); err != nil {
+			if err = b.updateSpecVolumes(updatedObj); err != nil {
 				return nil, err
 			}
 		}
 
-		if specsAreEqual, err := nestedUnstructuredComparison(&obj, updatedObj); err != nil {
+		if specsAreEqual, err := nestedUnstructuredComparison(&obj, updatedObj, "spec"); err != nil {
```

Comment:
This I assume was a bug in the original implementation, can you confirm?

Reply:
I have removed the extra argument.
be37227d54ed101c4e57a427fea93abc1cf54cdb